### PR TITLE
Switched Node.JS tests over to TypeScript.   Essentially, they are th…

### PR DESCRIPTION
**Pull request by BretJohnson**
…e same tests as before except now strongly typed.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@BretJohnson__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by stankovski** [comment]
@azuresdkci add to whitelist

**Comment by BretJohnson** [comment]
@amarzavery can you give this the once over & merge it.   Thanks.   The Travis CI hang, in the middle of running C# tests, looks unrelated.   Jenkins looks good.

File: AutoRest/Generators/NodeJS/NodeJS.Tests/typings/mocha/mocha.d.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
+// Type definitions for mocha 2.2.5
+// Project: http://mochajs.org/
```
Comment (amarzavery) [inline AutoRest/Generators/NodeJS/NodeJS.Tests/typings/mocha/mocha.d.ts]: Do we have to generate this typings whenever we update the mocha version?
At the same time, if we take a dependency on a new package then how do we generate typings for that package?
Is this Automated currently?

**Comment by amarzavery** [comment]
please pull from upstream dev and resolve conflicts

File: AutoRest/Generators/NodeJS/NodeJS.Tests/typings/mocha/mocha.d.ts (hunk 1 above)
Comment (BretJohnson) [inline AutoRest/Generators/NodeJS/NodeJS.Tests/typings/mocha/mocha.d.ts]: Yes, typings should be updated if changing the mocha version.   This should work to do that:
`tsd install mocha --save --overwrite`
For taking dependencies on a new package where want type checking this should do it:
`tsd install <package> --save`
See https://github.com/DefinitelyTyped/tsd for full doc.
Basically `tsd.json` stores the type dependencies (much like `package.json` for node dependencies) and `typings` is generated from that containing all the type files (like `node_modules`).

**Comment by BretJohnson** [comment]
I updated with the latest from dev.   It should be in good shape now.

**Comment by amarzavery** [comment]
@BretJohnson - Have you verified that these tests run in Visual Studio? We have a .sln file for generic and azure nodejs generator over here: 
https://github.com/Azure/autorest/blob/master/AutoRest/Generators/NodeJS/NodeJSGeneratorTests.sln

Debugging tests via visual studio is very handy when doing RCA when issues come up. Please make sure that the solution includes your .ts files and that you are able to run and debug tests via visual studio.

**Comment by amarzavery** [comment]
At the same time, I would suggest you to create a README.Md at this location https://github.com/Azure/autorest/tree/master/AutoRest/Generators/NodeJS. 

Please document the steps to generate the .ts files for scenarios when
- some one updates the mocha or should.js version
- some one adds additional dependencies, then how should they generate the .ts files for those dependencies.

**Comment by amarzavery** [comment]
Otherwise the .ts test files look good to me

**Pull request merged**